aws_list_all/generate_html.py

```diff
@@ -0,0 +1,271 @@
+import os
+import sys
+import webbrowser
+from collections import defaultdict
+from sys import exit, stderr
+
+from .listing import ResultListing
+
+DEFINE_HEADERBAR = """
+    .headerbar {overflow: hidden; background-color: #232F3E; position: fixed; top: 0; width: 100%;}
+    .headerbar a {float: left; color: #FFFFFF; font-size: 40px; font-family: arial;
+        padding: 10px 30px; text-decoration: none;}
+    .headerbar input[type=text] {margin-top: 12px; margin-right: 30px; border: none; float: right;}
+"""
+
+DEFINE_FOOTER = """
+    .footer {overflow: hidden; background-color: #232F3E; position: fixed; bottom: 0; width: 100%;}
+    .footer a {float: left; color: #f2f2f2; font-size: 12px; font-family: arial;
+        padding: 5px 5px; text-decoration: none;}
+"""
+
+DEFINE_TABLE = """
+    .aws-table th {border: none; border-collapse: collapse; border-radius: 10px; background-color: #f1f1f1; 
+        font-family: Arial; font-size: 20px; padding: 10px; text-align: center; table-layout:fixed;}\n
+    .aws-table td {border: none; border-collapse: collapse; border-radius: 10px; background-color: #f1f1f1; 
+        text-align: center; table-layout:fixed;}\n
+    .aws-table .service {border: none; border-collapse: collapse; font-family: Arial; 
+        font-size: 18px; text-align: center; table-layout:fixed; background-color: #f1f1f1;}\n
+"""
+
+DEFINE_SEARCHINPUT = """
+    #searchInput {\n'
+      background-position: 10px 10px;\n
+      background-repeat: no-repeat;\n
+      width: 300px;\n
+      font-size: 16px;\n
+      padding: 12px 20px 12px 40px;\n
+      border: 1px solid #ddd;\n
+      margin-bottom: 12px;\n
+    }\n
+    </style>\n
+    </head>\n
+"""
+DEFINE_POPUP = """
+    .popup {position: relative; display: inline-block; cursor: pointer; 
+        -webkit-user-select: none; -moz-user-select: none; -ms-user-select: none; user-select: none;}\n
+    .popup .popuptext {visibility: hidden; width: 160px; background-color: #555; color: #fff; 
+        text-align: center; border-radius: 6px; padding: 8px 0; position: absolute; z-index: 1; 
+        bottom: 125%; left: 50%; margin-left: -80px;}\n
+    .popup .popuptext::after {content: ""; position: absolute; top: 100%; left: 50%; margin-left: -5px; 
+        border-width: 5px; border-style: solid; border-color: #555 transparent transparent transparent;}\n
+    .popup .show {visibility: visible; -webkit-animation: fadeIn 1s; animation: fadeIn 1s;}\n
+    @-webkit-keyframes fadeIn {from {opacity: 0;} to {opacity: 1;}}\n
+    @keyframes fadeIn {from {opacity: 0;} to {opacity:1 ;}}\n
+"""
+
+
+def before_content(name):
+    """Open a file to write HTML-content in and return the original system output and file path"""
+    origout = sys.stdout
+    f = open(name, 'w')
+    sys.stdout = f
```

Comment:
I really don't like bending sys.stdout here like this. Ideally, all these functions should just add a string together (lines = []; lines.append instead of print; return "\n".join(lines))